### PR TITLE
Use pass by reference where possible

### PR DESCRIPTION
**Pull request by catornot**
there are probably more places where pass by reference can be used

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Just so that I'm on the same page, pass-by-reference > pass-by-value cause less copies, meaning slightly better performance, right? ^^"

**Comment by catornot** [comment]
well only for like large stuff, like`String`, `Vec`, etc . it would not make sense do it for stuff with `Copy` implemented like `i32`.

also `clone` slightly irritates me

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -488,17 +488,20 @@ pub fn launch_northstar_steam(
...
 pub fn check_northstar_running() -> bool {
     let s = sysinfo::System::new_all();
-    s.processes_by_name("NorthstarLauncher.exe")
+    let x = s
+        .processes_by_name("NorthstarLauncher.exe")
         .next()
         .is_some()
-        || s.processes_by_name("Titanfall2.exe").next().is_some()
+        || s.processes_by_name("Titanfall2.exe").next().is_some();
+    x
```
Comment (GeckoEidechse) [inline src-tauri/src/lib.rs:504]: Ditto

File: src-tauri/src/northstar/mod.rs [hunk 2]
```diff
@@ -5,24 +5,23 @@ use crate::{check_mod_version_number, constants::CORE_MODS};
...
     log::info!("{}", game_path);
 
     // TODO:
     // Check if NorthstarLauncher.exe exists and check its version number
     let profile_folder = "R2Northstar";
-    let initial_version_number = match check_mod_version_number(format!(
-        "{}/{}/mods/{}",
-        game_path, profile_folder, CORE_MODS[0]
+    let initial_version_number = match check_mod_version_number(&format!(
+        "{game_path}/{profile_folder}/mods/{}",
```
Comment (GeckoEidechse) [inline src-tauri/src/northstar/mod.rs:15]: Technically not a pass-by-reference related change but I'll allow it >:P

File: src-tauri/src/lib.rs [hunk 3]
```diff
@@ -488,17 +488,20 @@ pub fn launch_northstar_steam(
 
 pub fn check_origin_running() -> bool {
     let s = sysinfo::System::new_all();
-    s.processes_by_name("Origin.exe").next().is_some()
-        || s.processes_by_name("EADesktop.exe").next().is_some()
+    let x = s.processes_by_name("Origin.exe").next().is_some()
+        || s.processes_by_name("EADesktop.exe").next().is_some();
+    x
```
[the remark this answers is not in the document]
Comment (catornot) [reply src-tauri/src/lib.rs:493]: `error[E0597]` happened

[the remark this answers is not in the document]
Comment (GeckoEidechse) [reply src-tauri/src/lib.rs:493]: lol

**Pull request merged**